lib/storage/metadata/mongoclient/MongoClientInterface.ts

```diff
@@ -583,7 +587,13 @@ class MongoClientInterface {
         }, {
             upsert: true,
         })
-            .then(() => cb(null))
+            .then(result => {
+                if (result.matchedCount === 0 && result.modifiedCount === 0 && result.upsertedCount === 0) {
+                    log.debug('putBucketAttributes: failed to update bucket', { bucketName });
+                    return cb(errors.NoSuchBucket);
```

Comment:
nit:
- this is really a case which "should not" (but does) happen: may be best to keep it as InternalError, possibly customized with an indication of the "mongo" issue (`errors.InternalError.customize("mongo failed to match, update and upsert")`)
- maybe we can dedup this case of error detection after an update operation, by introducing a `processUpdateResult(result, cb)` function?